src/main/scala/firrtl/passes/AnnotateMemMacros.scala

```diff
@@ -72,6 +72,8 @@ object AnalysisUtils {
       if (nodeWidth == extractionWidth) getOrigin(connects, args.head) else e
     case DoPrim((PrimOps.AsUInt | PrimOps.AsSInt | PrimOps.AsClock), args, _, _) => 
       getOrigin(connects, args.head)
+    // Todo: It's not clear it's ok to call remove validifs before mem passes...
+    case ValidIf(cond, value, ClockType) => getOrigin(connects, value)
```

Comment:
There's nothing wrong with ValidIf existing in low Firrtl, eventually we might even use it in the emitter